pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/GeoPersistentReplicator.java

```diff
@@ -56,7 +63,115 @@ protected String getProducerName() {
     @Override
     protected CompletableFuture<Void> prepareCreateProducer() {
         if (brokerService.getPulsar().getConfig().isCreateTopicToRemoteClusterForReplication()) {
-            return CompletableFuture.completedFuture(null);
+            TopicName completeTopicName = TopicName.get(localTopicName);
+            TopicName baseTopicName;
+            if (completeTopicName.isPartitioned()) {
+                baseTopicName = TopicName.get(completeTopicName.getPartitionedTopicName());
+            } else {
+                baseTopicName = completeTopicName;
+            }
+            // Set useFallbackForNonPIP344Brokers to true when mix of PIP-344 and non-PIP-344 brokers are used, it
+            // can still work.
+            return client.getLookup().getPartitionedTopicMetadata(baseTopicName, false, true)
```

Comment:
Since we have `pulsarAdmin` object, can we use `pulsar-admin.topics.getPartitionedTopicMetadataAsync` here?

Reply:
Good point, I will refactor this logic.

Reply:
@poorbarcode This is an asynchronous method, but the keyword `async` is missing from its name.

Reply:
@poorbarcode Good, I should use pulsarAdmin instead of pulsarClient, this is better way.